### PR TITLE
SWATCH-3025: Create swatch-common-export module

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-3025

## Description
There is no porting functionality here, but only moving the export request handler/formatters logic into a common library that then will be reused by Quarkus and Spring Boot services (for now, only Spring Boot).

The new swatch-common-export module is framework agnostic. To achieve this, we use some delegates to operate with the actual export API and rbac API. This delegate must be provided by the Spring Boot/Quarkus services.

Captura de pantalla de 2024-10-14 17-00-56

## Testing
Only regression testing.
The existing logic is already covered by the existing junit tests ++ also via iqe tests.

### CODE REVIEW

**Comment by Sgitario** [comment]
@kartikshahc @awood this PR has the changes we have discussed. Any of you is volunteer to take the DEV review?
Note that I renamed the ExportService to ExportRequestHandler and ExportConsumer to ExportRequestConsumer to better clarity.

File: src/main/java/org/candlepin/subscriptions/tally/export/InstancesDataExporterService.java [hunk 1]
```diff
@@ -26,6 +26,11 @@
 
 import com.redhat.swatch.configuration.registry.MetricId;
 import com.redhat.swatch.configuration.registry.ProductId;
+import com.redhat.swatch.export.DataExporterService;
+import com.redhat.swatch.export.DataMapperService;
+import com.redhat.swatch.export.ExportServiceException;
+import com.redhat.swatch.export.ExportServiceRequest;
+// NOTE(khowell): this couples our export implementation to the v1 REST API
```
Comment (awood) [inline src/main/java/org/candlepin/subscriptions/tally/export/InstancesDataExporterService.java:33]: This comment needs to move down to line 53 now I believe.  Above `import org.candlepin.subscriptions.utilization.api.v1.model.ReportCategory;`

File: swatch-common-export/src/main/java/com/redhat/swatch/export/ExportRequestHandler.java [hunk 2]
```diff
@@ -18,82 +18,43 @@
...
 @Slf4j
-public class ExportSubscriptionListener extends SeekableKafkaConsumer {
+@AllArgsConstructor
+public class ExportRequestHandler {
 
   public static final String ADMIN_ROLE = ":*:*";
   public static final String SWATCH_APP = "subscriptions";
   public static final String MISSING_PERMISSIONS = "Insufficient permission";
+  public static final int INTERNAL_ERROR = 500;
+  public static final int FORBIDDEN_STATUS = 403;
```
Comment (awood) [inline swatch-common-export/src/main/java/com/redhat/swatch/export/ExportRequestHandler.java:41]: How do you feel about dropping these and just using `HttpURLConnection.HTTP_FORBIDDEN`, etc?

**Review by awood — CHANGES_REQUESTED** [review]
Looks great.  If you could just correct that one mislaid comment; otherwise, excellent.

File: src/main/java/org/candlepin/subscriptions/tally/export/InstancesDataExporterService.java (hunk 1 above)
Comment (Sgitario) [reply src/main/java/org/candlepin/subscriptions/tally/export/InstancesDataExporterService.java:33]: yes, thanks for spotting.

File: swatch-common-export/src/main/java/com/redhat/swatch/export/ExportRequestHandler.java (hunk 2 above)
Comment (Sgitario) [reply swatch-common-export/src/main/java/com/redhat/swatch/export/ExportRequestHandler.java:41]: +1

**Comment by Sgitario** [comment]
@awood I've already updated the PR after your comments

**Comment by Sgitario** [comment]
Merging after applying the suggested changes. Failures are unrelated.

**Pull request merged**